dbg.h

```diff
@@ -721,6 +772,39 @@ pretty_print(std::ostream& stream, const Container& value) {
   return true;
 }
 
+template <typename ContainerAdapter>
+inline typename std::enable_if<
+    detail::is_container_adapter<const ContainerAdapter&>::value,
+    bool>::type
+pretty_print(std::ostream& stream, ContainerAdapter value) {
+  stream << "{";
+  const size_t size = detail::size(value);
+  const size_t n = std::min(size_t{10}, size);
+
+  std::vector<typename ContainerAdapter::value_type> elements;
+  elements.reserve(n);
+  for (size_t i = 0; i < n; ++i) {
+    elements.push_back(std::move(detail::pop(value)));
```

Comment:
That `std::move` is also unnecessary.
```suggestion
    elements.push_back(detail::pop(value));
```

Reply:
Even here, `vector::push_back` has overloads for l-value and r-value reference and from what I know without explicit `move` operation there will be an unnecessary copy.

Reply:
I just made suggested changes locally and there seems to be a lot of unnecessary copies without `std::move` operation. Here is a working example to reproduce.
```cpp
struct MyStruct {
  MyStruct() = default;

  MyStruct(const MyStruct&) noexcept {
    std::cerr << "copied" << std::endl;
  }

  MyStruct(MyStruct&&) noexcept {
    // std::cerr << "moved" << std::endl;
  }

  friend void swap(MyStruct& a, MyStruct& b)
  {
  }
};

std::ostream& operator<<(std::ostream& out, const MyStruct&) {

    out << "(MyStruct)";
    return out;
}

void main() {
    vector<MyStruct> v;
    v.emplace_back();
    v.emplace_back();
    v.emplace_back();
    // v now has 3 elements
    stack<vector<MyStruct>> st;
    st.push(v);  // 3 copies
    st.push(v);  // 3 copies

    dbg("calling dbg");
    dbg(st); // 12 copies 
}
```

Ideally we should see only 6 copies since the adapter should be copied entirely for inspecting elements.

Reply:
... but `detail::pop` returns a value, correct? So this will move in any case: https://godbolt.org/z/odfGKMq5W

Reply:
My bad. You are right. `move` operation in `push_back` is redundant. Corrected.